### ISSUE TITLE
Layout/SpaceInsideArrayLiteralBrackets crashes

### ISSUE THREAD

**Issue by akimd** [rubocop/rubocop#14157]
AFAICT, this was not reported yet.

## Expected behavior

I have tons of crashes on file such as the following excerpt:
```ruby
case lhs
in Type::CConj[*head, tail]
in Type::CDef[term_var, type_scheme, constraint]
in Type::CExist[type_var, constraint]
end
```

## Actual behavior

```
$ bundle exec rubocop rubocop/space_inside_array_literal_bracket.rb
Inspecting 1 file
An error occurred while Layout/SpaceInsideArrayLiteralBrackets cop was inspecting rubocop/space_inside_array_literal_bracket.rb:2:15.
To see the complete backtrace run rubocop -d.
An error occurred while Layout/SpaceInsideArrayLiteralBrackets cop was inspecting rubocop/space_inside_array_literal_bracket.rb:3:14.
To see the complete backtrace run rubocop -d.
An error occurred while Layout/SpaceInsideArrayLiteralBrackets cop was inspecting rubocop/space_inside_array_literal_bracket.rb:4:16.
To see the complete backtrace run rubocop -d.
E

Offenses:

rubocop/space_inside_array_literal_bracket.rb:1:1: C: [Correctable] Style/FrozenStringLiteralComment: Missing frozen string literal comment.
case lhs
^
rubocop/space_inside_array_literal_bracket.rb:2:1: E: Lint/EmptyInPattern: Avoid in branches without a body.
in Type::CConj[*head, tail]
^^^^^^^^^^^^^^^^^^^^^^^^^^^
rubocop/space_inside_array_literal_bracket.rb:3:1: E: Lint/EmptyInPattern: Avoid in branches without a body.
in Type::CDef[term_var, type_scheme, constraint]
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
rubocop/space_inside_array_literal_bracket.rb:4:1: E: Lint/EmptyInPattern: Avoid in branches without a body.
in Type::CExist[type_var, constraint]
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

1 file inspected, 4 offenses detected, 1 offense autocorrectable

3 errors occurred:
An error occurred while Layout/SpaceInsideArrayLiteralBrackets cop was inspecting rubocop/space_inside_array_literal_bracket.rb:2:15.
An error occurred while Layout/SpaceInsideArrayLiteralBrackets cop was inspecting rubocop/space_inside_array_literal_bracket.rb:3:14.
An error occurred while Layout/SpaceInsideArrayLiteralBrackets cop was inspecting rubocop/space_inside_array_literal_bracket.rb:4:16.
Errors are usually caused by RuboCop bugs.
Please, update to the latest RuboCop version if not already in use, and report a bug if the issue still occurs on this version.
https://github.com/rubocop/rubocop/issues

Mention the following information in the issue report:
1.75.5 (using Parser 3.3.8.0, rubocop-ast 1.44.1, analyzing as Ruby 2.7, running on ruby 3.3.8) [arm64-darwin24]
```

Note that I do not understand why it says "analyzing as Ruby 2.7" since I require 3.3, and indeed:

## RuboCop version
```
$ bundle exec rubocop -V
1.75.5 (using Parser 3.3.8.0, Prism 1.4.0, rubocop-ast 1.44.1, analyzing as Ruby 3.3, running on ruby 3.3.8) [arm64-darwin24]
  - rubocop-minitest 0.38.0
  - rubocop-performance 1.25.0
```

Cheers!

**Comment by koic** [comment]
Thank you for the feedback. This issue was resolved in https://github.com/rubocop/rubocop/pull/14152, but the fix has not been released yet.
